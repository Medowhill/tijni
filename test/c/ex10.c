#include <stdlib.h>

void f(int *p) {
  *p = 0;
}

void g(int *p) {
  free(p);
}

void h(int *p, int *q, int x) {
  if (x) {
    f(p);
    g(q);
  } else {
    g(p);
    f(q);
  }
}
