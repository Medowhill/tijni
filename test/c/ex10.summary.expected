{
  "f": {
    "params": [ { "name": "%p(f)", "used": true, "freed": false } ],
    "ret": [],
    "env": { "%p(f)": [ "'%p(f)" ] }
  },
  "g": {
    "params": [ { "name": "%p(g)", "used": false, "freed": true } ],
    "ret": [],
    "env": {
      "%p(g)": [ "'%p(g)" ],
      "%0(g)": [ "'%p(g)" ]
    }
  },
  "h": {
    "params": [ { "name": "%p(h)", "used": true, "freed": true }, { "name": "%q(h)", "used": true, "freed": true }, { "name": "%x(h)", "used": false, "freed": false } ],
    "ret": [],
    "env": {
      "%p(f)": [ "'%p(h)", "'%q(h)" ],
      "%p(g)": [ "'%p(h)", "'%q(h)" ],
      "%0(g)": [ "'%p(h)", "'%q(h)" ],
      "%p(h)": [ "'%p(h)" ],
      "%q(h)": [ "'%q(h)" ],
      "%x(h)": [ "'%x(h)" ],
      "%tobool(h)": []
    }
  }
}